transcendence-app/src/game/game.controller.ts

```diff
@@ -0,0 +1,94 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  HttpCode,
+  HttpStatus,
+  Post,
+  Query,
+  ServiceUnavailableException,
+  UnprocessableEntityException,
+  UseGuards,
+} from '@nestjs/common';
+import { GameService } from './game.service';
+import { TwoFactorAuthenticatedGuard } from '../shared/guards/two-factor-authenticated.guard';
+import {
+  ApiBadRequestResponse,
+  ApiCreatedResponse,
+  ApiForbiddenResponse,
+  ApiNotFoundResponse,
+  ApiOkResponse,
+  ApiParam,
+  ApiServiceUnavailableResponse,
+  ApiTags,
+  ApiUnprocessableEntityResponse,
+} from '@nestjs/swagger';
+import { Game } from './infrastructure/db/game.entity';
+import { MAX_ENTRIES_PER_PAGE } from '../shared/constants';
+import { PaginationWithSearchQueryDto } from '../shared/dtos/pagination-with-search.query.dto';
+import { CreateGameDto } from './dto/create-game.dto';
+
+@Controller()
+@UseGuards(TwoFactorAuthenticatedGuard)
+@ApiTags('game')
+@ApiForbiddenResponse({ description: 'Forbidden' })
+export class GameController {
+  constructor(private gameService: GameService) {}
+
+  @Post('game')
+  @ApiCreatedResponse({ description: 'Add a game', type: Game })
+  @ApiBadRequestResponse({ description: 'Bad Request' })
+  @ApiUnprocessableEntityResponse({ description: 'Unprocessable entity' })
+  async addGame(@Body() game: CreateGameDto): Promise<Game> {
+    const savedGame = await this.gameService.addGame(game);
+
+    if (!savedGame) {
+      throw new UnprocessableEntityException();
+    }
+    return savedGame;
+  }
+
+  @Delete('game')
+  @HttpCode(HttpStatus.NO_CONTENT)
+  @ApiParam({ name: 'gameId', type: String })
+  @ApiBadRequestResponse({ description: 'Bad Request' })
+  @ApiNotFoundResponse({ description: 'Not Found' })
+  @ApiServiceUnavailableResponse({ description: 'Service Unavailable' })
+  async removeGame(gameId: string): Promise<void> {
+    return this.gameService.deleteGame(gameId);
+  }
+
+  @Get('game')
+  @ApiOkResponse({
+    description: `Get a game`,
+    type: [Game],
+  })
+  @ApiParam({ name: 'gameId', type: String })
+  @ApiBadRequestResponse({ description: 'Bad Request' })
+  @ApiServiceUnavailableResponse({ description: 'Service unavailable' })
+  async getGame(gameId: string): Promise<Game> {
+    const game = await this.gameService.retrieveGameWithId(gameId);
+    if (!game) {
+      throw new ServiceUnavailableException();
+    }
+    return game;
+  }
+
+  @Get('games')
+  @ApiOkResponse({
+    description: `Lists all games (max ${MAX_ENTRIES_PER_PAGE})`,
```

Comment:
I don't think we need this endpoint like this. To list them in the `/play`  page, we need the ongoing games, and those ones are not going to be in the database, AFAIK.
We need to retrieve games for a particular player in order to retrieve the stats for particular user page, but not the games in bulk. We could reuse the 'search' field of the pagination query to find the available data from this endpoint as is, but then again the '%search%' percents will make it difficult.

In my opinion we should pass to this endpoint the username as a path parameter.

Reply:
You are right. We can retrieve the games for a particular user and enable the search for the second user, the opponent, or disable search.
I will change it

Reply:
thought: We could leave this as is because it works and maybe (in the future) we want to create a page to list all the games played.

And as Damrod suggests, we could create additional endpoints to list all the games the authenticated user has played (`/games/user` ?) or to list all the games for a particular user (`/games/:userId` ?). We could do this in a follow up PR.

Reply:
These two comments for removing the I from iLike and removing the percents (%) were only valid to retrieve games for a particular user. If we're retrieving the games for any user matching the search, then we still need the i in iLike and the two percents (%). Think about what returns what, and stay consistent.